meta-mender-core/classes/mender-artifactcapsule.bbclass

```diff
@@ -0,0 +1,151 @@
+        provides += " --provides " + capsule_type + "." + guid + ".version:" + splt[1]
+        provides += " --provides " + capsule_type + "." + guid + ".name:" + splt[0]
+    return provides
+
+def get_mender_capsule_artifact_depends(d):
+    # This function creates the depends format required by mender artifact to
+    # generate the capsule artifact.
+    # MENDER_ARTIFACT_CAPSULE_DEPENDS needs to be used in the following way.
+    #MENDER_ARTIFACT_CAPSULE_DEPENDS[e2bb9c06-70e9-4b14-97a3-5a7913176111] = "2"
+    #MENDER_ARTIFACT_CAPSULE_DEPENDS[e2bb9c06-70e9-4b14-97a3-5a7913176222] = "2"
+    # This gets converted to the  following depends format:
+    # "uefi-firmware.e2bb9c06-70e9-4b14-97a3-5a7913176111.version":"2"
+    # "uefi-firmware.e2bb9c06-70e9-4b14-97a3-5a7913176222.version":"2"
+    guids = d.getVarFlags("MENDER_ARTIFACT_CAPSULE_DEPENDS")
+    capsule_type = d.getVar("CAPSULE_TYPE", expand=True)
+    if guids is None:
+        return
+    depends = ""
+    for guid in guids:
+        depends += " --depends " + capsule_type + "." + guid + ".version:" + guids[guid]
+    return depends
+
+def get_mender_capsule_artifact_clears_provides(d):
+    # This function creates the clears provides format required by mender artifact
+    # to generate the capsule artifact.
+    guids = d.getVarFlags("MENDER_ARTIFACT_CAPSULE_PROVIDES")
+    capsule_type = d.getVar("CAPSULE_TYPE", expand=True)
+    if guids is None:
+        return
+    clears_provides = ""
+    for guid in guids:
+        clears_provides += " --clears-provides " + capsule_type + "." + guid + ".version"
+    return clears_provides
+
+MENDER_PROVIDES = "${@get_mender_capsule_artifact_provides(d)}"
+MENDER_CLEARS_PROVIDES = "${@get_mender_capsule_artifact_clears_provides(d)}"
+MENDER_DEPENDS = "${@get_mender_capsule_artifact_depends(d)}"
+
+# --------------------------- END OF CONFIGURATION -----------------------------
+do_image_mender_capsule[depends] += "mender-artifact-native:do_populate_sysroot"
+
+IMAGE_CMD:mender-capsule () {
+    set -x
+
+    if [ -z "${MENDER_ARTIFACT_CAPSULE_NAME}" ]; then
+        bbfatal "Need to define MENDER_ARTIFACT_CAPSULE_NAME variable."
+    fi
+
+    if [ -z "${UEFI_CAPSULE}" ]; then
+        bbfatal "Need to define UEFI_CAPSULE variable."
+    fi
+
+    if [ -z "${MENDER_DEVICE_TYPES_COMPATIBLE}" ]; then
+        bbfatal "MENDER_DEVICE_TYPES_COMPATIBLE variable cannot be empty."
+    fi
+
+    if [ "${MENDER_PROVIDES}" = "None" ]; then
+        bbfatal "MENDER_ARTIFACT_CAPSULE_PROVIDES variable cannot be empty."
+    fi
+
+    extra_args=""
+
+    for dev in ${MENDER_DEVICE_TYPES_COMPATIBLE}; do
+        extra_args="$extra_args -t $dev"
+    done
+
+    if [ -n "${MENDER_ARTIFACT_SIGNING_KEY}" ]; then
+        extra_args="$extra_args -k ${MENDER_ARTIFACT_SIGNING_KEY}"
+    fi
+
+    if mender-artifact write rootfs-image --help | grep -e '-u FILE'; then
+        image_flag="-u"
+    else
+        image_flag="-f"
+    fi
```

Comment:
Cleanup: This conditional can be removed, even from the original location you grabbed it from if you feel like cleaning up some more. The version of mender-artifact which required the `-u` option is not available in the layer anymore (< 3.0).